web/src/lib/components/onboarding-page/onboarding-card.svelte

```diff
@@ -1,15 +1,32 @@
 <script lang="ts">
   import Icon from '$lib/components/elements/icon.svelte';
+  import { Button } from '@immich/ui';
+  import { mdiArrowLeft, mdiArrowRight, mdiCheck } from '@mdi/js';
   import type { Snippet } from 'svelte';
+  import { t, type Translations } from 'svelte-i18n';
   import { fade } from 'svelte/transition';
 
   interface Props {
-    title?: string | undefined;
+    title?: Translations | undefined;
     icon?: string | undefined;
     children?: Snippet;
+    previousTitle?: Translations | undefined;
+    nextTitle?: Translations | undefined;
```

Comment:
I don't think the titles should be a translation. The component should just accept the titles as strings, and the page/component that's actually using the component takes care of i18n. I.e. it then sets `title={$t('my_key')}`

Reply:
Hmmm I would've expected $t to be stateful.

Reply:
The settings page also just saves it and it works without a reload, no? Couldn't you just save the language on select?

Reply:
The issue that I would run into is that everything in the card would change *except* the button labels because they are using pre-translated strings from the array instead of translating them when the state changes.